src/QsCompiler/LlvmBindings/BitcodeModule.cs

```diff
@@ -383,6 +422,77 @@ public IrFunction CreateFunction(string name, IFunctionType signature)
             return Value.FromHandle<IrFunction>(valueRef)!;
         }
 
+        /// <summary>Creates a Function definition with Debug information</summary>
+        /// <param name="scope">Containing scope for the function</param>
+        /// <param name="name">Name of the function in source language form</param>
+        /// <param name="mangledName">Mangled linker visible name of the function (may be same as <paramref name="name"/> if mangling not required by source language</param>
+        /// <param name="file">File containing the function definition</param>
+        /// <param name="line">line number of the function definition (1-based)</param>
+        /// <param name="signature">LLVM Function type for the signature of the function</param>
+        /// <param name="isLocalToUnit">Flag to indicate if this function is local to the compilation unit</param>
+        /// <param name="isDefinition">Flag to indicate if this is a definition</param>
+        /// <param name="scopeLine">First line of the function's outermost scope, this may not be the same as the first line of the function definition due to source formatting (1-based)</param>
+        /// <param name="debugFlags">Additional flags describing this function</param>
+        /// <param name="isOptimized">Flag to indicate if this function is optimized</param>
+        /// <param name="dIBuilder"><see cref="DebugInfoBuilder"/>to use when creating debug info</param>
+        /// <returns>Function described by the arguments</returns>
+        /// <remarks>
+        /// If a matching function already exists it is returned, and therefore the returned
+        /// <see cref="IrFunction"/> may have a body and additional attributes. If a function of
+        /// the same name exists with a different signature an exception is thrown as LLVM does
+        /// not perform any function overloading. If the matching function already has a DISubProgram,
+        /// it is returned directly. If not, the given information is used to create a DISubProgram
+        /// </remarks>
+        public IrFunction CreateFunction(
+            DIScope? scope,
+            string name,
+            string? mangledName,
+            DIFile? file,
+            uint line,
+            DebugFunctionType signature,
+            bool isLocalToUnit,
+            bool isDefinition,
+            uint scopeLine,
+            DebugInfoFlags debugFlags,
+            bool isOptimized,
+            DebugInfoBuilder dIBuilder)
+        {
+            this.ThrowIfDisposed();
```

Comment:
I see `this.ThrowIfDisposed()` in multiple files. What is the purpose of this call?

Reply:
BitcodeModule has a member called isDisposed that can be set to true to "dispose" of the module. This is a safety check that was already used throughout our codebase in the places where LLVM.NET had different safety checks.